src/zenml/utils/source_utils.py

```diff
@@ -820,3 +824,198 @@ def _should_load_from_main_module(source: Source) -> bool:
         return False
 
     return resolved_main_module == source.module
+
+
+ObjectType = Union[
+    Type[Any],
+    Callable[..., Any],
+    ModuleType,
+    FunctionType,
+    BuiltinFunctionType,
+]
+
+
+class SourceOrObject:
```

Comment:
I love this class and the way you made it work with pydantic!

I think that class might make more sense in `zenml.config.source`, wdyt?

Reply:
Done